src/algorithm/navigation_algorithm.py

```diff
@@ -0,0 +1,184 @@
+import open3d as o3d
+import numpy as np
+import time
+from queue import PriorityQueue
+import random
+import heapq
+import matplotlib.pyplot as plt
+
+
+class AStarGrid:
+    def __init__(self, rows, cols, start_x, start_y, end_x, end_y):
+        self.rows = rows
+        self.cols = cols
+        self.spacing = 1.0
+        self.grid = np.zeros((rows, cols), dtype=int)
+        self.start = (start_x, start_y)
+        self.goal = (end_x, end_y)
+
+    def generate_random_grid(self, rows, cols):
```

Comment:
przydałoby się zęby to nie był totalnie random tylko żeby to była mapa gdzie są jakieś "obiekty" skupiska "0"  a reszta 1 bardziej to odzwierciedli rzeczywistość przykładowy kod:  
```python
def generate_obstacle_map(size, num_obstacles, max_obstacle_size):
    grid = np.ones((size, size), dtype=int)  # Initialize grid with open space (1)

    for _ in range(num_obstacles):
        # Randomly place a rectangular obstacle
        obstacle_width = np.random.randint(10, max_obstacle_size)
        obstacle_height = np.random.randint(10, max_obstacle_size)
        x_start = np.random.randint(0, size - obstacle_width)
        y_start = np.random.randint(0, size - obstacle_height)

        grid[x_start:x_start + obstacle_width, y_start:y_start + obstacle_height] = 0

    return grid


def visualize_grid(grid):
    plt.figure(figsize=(10, 10))
    plt.imshow(grid, cmap='gray', interpolation='none')
    plt.title("Obstacle Map")
    plt.colorbar()
    plt.show()


# Parameters
size = 5000
num_obstacles = 200  # Number of obstacles
max_obstacle_size = 200  # Maximum size of any obstacle

# Generate and visualize the map
obstacle_map = generate_obstacle_map(size, num_obstacles, max_obstacle_size)
visualize_grid(obstacle_map)
```
![image](https://github.com/knmlprz/TrailblazerML/assets/101070249/d17c17c2-2c40-49b8-9392-1a39c12288f1)